python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -2830,10 +2830,13 @@ def _update_asset_check_evaluation(self, event: EventLogEntry, event_id: Optiona
                     ),
                 )
             ).rowcount
-        if rows_updated != 1:
+
+        # 0 isn't normally expected, but occurs with the external instance of step launchers where
+        # they don't have planned events.
+        if rows_updated > 1:
```

Comment:
Since the writes have all happened already, is hard error here the right thing to do? 

What motivated this defense originally? Do we think this condition is likely to occur or particularly problematic if it does? 

Just trying to think through if we should drop this entirely or warn instead or something

Reply:
It was helpful during development when there were issues generating the execution plan that would cause planned events to not be logged. It also caught the bug I introduced in https://github.com/dagster-io/dagster/pull/19443...

It is problematic because of how the summary table works. The planned events create a row and the evaluation events update the row. If there's no planned event, we won't display the check result in the UI. If there's multiple, we'll show duplicate evaluations.

Looking at it now, the intent was to catch duplicate planned events early with a unique index. We have

```python
db.Index(
    "idx_asset_check_executions_unique",
    AssetCheckExecutionsTable.c.asset_key,
    AssetCheckExecutionsTable.c.check_name,
    AssetCheckExecutionsTable.c.run_id,
    AssetCheckExecutionsTable.c.partition,
    unique=True,
```

but this didn't catch the duplicate here because partition was null, which postgres considers unique. I'm going to put up a migration to add `NULLS NOT DISTINCT` which should make this invariant impossible (at least once oss users have run the migration). I still don't mind having it, in case the unique index got out of sync with the filter we're using to find rows to update.

Reply:
Actually `NULLS NOT DISTINCT` didn't exist until postgres 15+, do we have a min pg version on oss

Reply:
>  do we have a min pg version on oss

I am not sure but 15 is pretty recent so i don't  think we can enforce its use